state/application.go

```diff
@@ -1741,7 +1735,7 @@ func (a *Application) addUnitOpsWithCons(args applicationAddUnitOpsArgs) (string
 func (a *Application) addUnitSubordinateCharmProfileOp(subName, principalName string, profile *charm.LXDProfile) ([]txn.Op, error) {
 	// Because this is not part of a charm upgrade path, it is okay to
 	// short circuit here.
-	if profile == nil || (profile != nil && profile.Empty()) {
```

Comment:
Another tautology I couldn't resist removing.

Sorry for the noise from these - it's just in the 1 file/commit.